kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/KubernetesListOperationsImpl.java

```diff
@@ -152,7 +152,7 @@ public Boolean delete(List<KubernetesList> lists) {
     for (KubernetesList list : lists) {
       for (HasMetadata item : list.getItems()) {
         ResourceHandler<HasMetadata, HasMetadataVisitiableBuilder> handler = Handlers.get(item.getKind(), item.getApiVersion());
-        if (!handler.delete(client, config, namespace, item)) {
+        if (!handler.delete(client, config, namespace, false, item)) {
```

Comment:
I don't have a good understanding of this code, but is it possible that this PR (and possibly this line) changed the default value for `cascading` from `true` to `false` in some scenarios? I just upgraded from 4.1.3 to 4.3.0 and some things that were previously garbage collected are left orphaned now.

Reply:
Yes, it might be possible. Let me check whenever I get time. Sorry for the inconvenience!

Reply:
Speculated to have caused [JENKINS-58306](https://issues.jenkins-ci.org/browse/JENKINS-58306).